imagebuildah/util.go

```diff
@@ -1,9 +1,92 @@
 package imagebuildah
 
 import (
+	"archive/tar"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
 	"github.com/containers/buildah"
+	digest "github.com/opencontainers/go-digest"
 )
 
+type MountInfo struct {
+	Type   string
+	Source string
+	From   string
+}
+
+// Consumes mount flag in format of `--mount=type=bind,src=/path,from=image` and
+// return MountInfo with values, otherwise values are empty if keys are not present in the option.
+func GetFromAndSourceKeysFromMountFlag(mount string) MountInfo {
+	tokens := strings.Split(mount, ",")
+	source := ""
+	from := ""
+	mountType := ""
+	for _, option := range tokens {
+		if strings.Contains(option, "=") {
+			optionSplit := strings.Split(option, "=")
+			if optionSplit[0] == "src" || optionSplit[0] == "source" {
+				source = optionSplit[1]
+			}
+			if optionSplit[0] == "from" {
+				from = optionSplit[1]
+			}
+			if optionSplit[1] == "type" {
+				mountType = optionSplit[2]
+			}
+		}
+	}
+	return MountInfo{Source: source, From: from, Type: mountType}
+}
+
+// GeneratePathChecksum generates the SHA-256 checksum for a file or a directory.
+func GeneratePathChecksum(sourcePath string) (string, error) {
+	digester := digest.SHA256.Digester()
+	tarWriter := tar.NewWriter(digester.Hash())
+
+	err := filepath.Walk(sourcePath, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		header, err := tar.FileInfoHeader(info, "")
```

Comment:
This second parameter is still incorrect for cases where, I think, `info.Mode().Type()` is `fs.ModeSymlink`.  The integration test should include at least one symbolic link in the test directory, which is overwritten using `ln -sf` between invocations.